Lib/fontTools/cffLib.py

```diff
@@ -27,21 +32,38 @@ def __init__(self):
 		pass
 
 	def decompile(self, file, otFont):
-		sstruct.unpack(cffHeaderFormat, file.read(4), self)
-		assert self.major == 1 and self.minor == 0, \
-				"unknown CFF format: %d.%d" % (self.major, self.minor)
-
-		file.seek(self.hdrSize)
-		self.fontNames = list(Index(file))
-		self.topDictIndex = TopDictIndex(file)
-		self.strings = IndexedStrings(file)
-		self.GlobalSubrs = GlobalSubrsIndex(file)
-		self.topDictIndex.strings = self.strings
-		self.topDictIndex.GlobalSubrs = self.GlobalSubrs
+		global isCFF2
```

Comment:
More accurately, I would need to establish as parent-child relationship that does not now exist throughout the entire tree of structures. In some cases this can be done with adding a new argument to carry the context, in others the subclasses would need to be able to access parent classes.